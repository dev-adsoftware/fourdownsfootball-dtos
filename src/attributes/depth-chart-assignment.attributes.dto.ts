import { IsNumber, IsString } from 'class-validator';
import { Dto } from '../dto';

export class DepthChartAssignmentAttributes extends Dto {
  @IsString()
  playerId: string;

  @IsString()
  fromPosition: string;

  @IsNumber()
  fromRank: number;

  @IsString()
  toPosition: string;

  @IsNumber()
  toRank: number;
}
