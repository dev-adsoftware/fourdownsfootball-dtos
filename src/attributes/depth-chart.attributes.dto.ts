import { IsString } from 'class-validator';
import { Dto } from '../dto';

export class DepthChartAttributes extends Dto {
  @IsString({ each: true })
  qb: string[];

  @IsString({ each: true })
  rb: string[];

  @IsString({ each: true })
  fb: string[];

  @IsString({ each: true })
  wr: string[];

  @IsString({ each: true })
  te: string[];

  @IsString({ each: true })
  lt: string[];

  @IsString({ each: true })
  lg: string[];

  @IsString({ each: true })
  c: string[];

  @IsString({ each: true })
  rg: string[];

  @IsString({ each: true })
  rt: string[];

  @IsString({ each: true })
  dt: string[];

  @IsString({ each: true })
  de: string[];

  @IsString({ each: true })
  ilb: string[];

  @IsString({ each: true })
  olb: string[];

  @IsString({ each: true })
  cb: string[];

  @IsString({ each: true })
  s: string[];

  @IsString({ each: true })
  k: string[];

  @IsString({ each: true })
  p: string[];

  @IsString({ each: true })
  kr: string[];

  @IsString({ each: true })
  pr: string[];

  @IsString({ each: true })
  ksp: string[];

  @IsString({ each: true })
  krsp: string[];

  @IsString({ each: true })
  prsp: string[];

  @IsString({ each: true })
  psp: string[];
}
