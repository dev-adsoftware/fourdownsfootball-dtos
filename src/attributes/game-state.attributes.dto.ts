import { IsBoolean, IsEnum, IsNumber, IsString } from 'class-validator';
import { Dto } from '../dto';
import { DirectionChoices } from '../types/direction-choices.enum';
import { GameState } from '../types/game-state.enum';

export class GameStateAttributes extends Dto {
  @IsEnum(GameState)
  state: GameState;

  @IsString()
  actorId: string;

  @IsString()
  offense = 'none';

  @IsEnum(DirectionChoices)
  direction: DirectionChoices = DirectionChoices.South;

  @IsNumber()
  homeScore = 0;

  @IsNumber()
  awayScore = 0;

  @IsNumber()
  period = 1;

  @IsNumber()
  gameClock = 15 * 60;

  @IsNumber()
  ballOn = 0;

  @IsNumber()
  down = 0;

  @IsNumber()
  toGo = 0;

  @IsBoolean()
  isClockRunning = false;
}
