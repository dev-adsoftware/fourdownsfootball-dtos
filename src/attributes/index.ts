export * from './player.attributes.dto';
export * from './player-assignment.attributes.dto';
export * from './team.attributes.dto';
