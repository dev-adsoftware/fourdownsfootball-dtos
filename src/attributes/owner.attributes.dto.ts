import { IsString } from 'class-validator';
import { Dto } from '../dto';

export class OwnerAttributes extends Dto {
  @IsString()
  username: string;
}
