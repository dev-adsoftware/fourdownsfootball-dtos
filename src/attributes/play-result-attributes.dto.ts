import {
  IsArray,
  IsBoolean,
  IsNumber,
  IsOptional,
  IsString,
} from 'class-validator';
import { Dto } from '../dto';

export class PlayResultAttributes extends Dto {
  @IsNumber()
  @IsOptional()
  kickDistance?: number;

  @IsBoolean()
  @IsOptional()
  returned?: boolean;

  @IsNumber()
  @IsOptional()
  returnDistance?: number;

  @IsString()
  @IsOptional()
  kickedBy?: string;

  @IsString()
  @IsOptional()
  returnedBy?: string;

  @IsString()
  @IsOptional()
  tackledBy?: string;

  @IsArray()
  @IsOptional()
  staminaReductions?: { id: string; value: number }[];
}
