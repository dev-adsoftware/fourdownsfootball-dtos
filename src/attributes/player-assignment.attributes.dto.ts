import { IsNumberString, IsString } from 'class-validator';
import { Dto } from '../dto';

export class PlayerAssignmentAttributes extends Dto {
  @IsString()
  id: string;

  @IsNumberString()
  sequence: string;
}
