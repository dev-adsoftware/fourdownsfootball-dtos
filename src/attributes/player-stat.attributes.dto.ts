import { IsDivisibleBy, IsNumber } from 'class-validator';
import { Dto } from '../dto';

export class PlayerStatAttributes extends Dto {
  @IsNumber()
  @IsDivisibleBy(1)
  plays: number;

  @IsNumber()
  @IsDivisibleBy(1)
  kickoffs: number;

  @IsNumber()
  @IsDivisibleBy(1)
  kickoffYds: number;

  @IsNumber()
  @IsDivisibleBy(1)
  kickoffTouchbacks: number;

  @IsNumber()
  @IsDivisibleBy(1)
  kickReturns: number;

  @IsNumber()
  @IsDivisibleBy(1)
  kickReturnYds: number;

  @IsNumber()
  @IsDivisibleBy(1)
  kickReturnTDs: number;

  @IsNumber()
  @IsDivisibleBy(1)
  tackles: number;
}
