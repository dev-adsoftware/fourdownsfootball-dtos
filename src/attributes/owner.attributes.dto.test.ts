import { OwnerAttributes } from './owner.attributes.dto';

describe('given: missing data', () => {
  describe('when: I initialize', () => {
    it('then: error matched snapshot', async () => {
      expect(() =>
        new OwnerAttributes().init({}),
      ).toThrowErrorMatchingSnapshot();
    });
  });
});
