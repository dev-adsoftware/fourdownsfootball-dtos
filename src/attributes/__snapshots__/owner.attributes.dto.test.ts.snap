// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`given: missing data when: I initialize then: error matched snapshot 1`] = `"OwnerAttributes: username must be a string"`;
